src/core/lib/support/murmur_hash.cc

```diff
@@ -45,7 +45,7 @@ uint32_t gpr_murmur_hash3(const void *key, size_t len, uint32_t seed) {
 
   /* body */
   for (i = -(int)nblocks; i; i++) {
-    memcpy(&k1, blocks + i, sizeof(uint32_t));
+    memcpy(&k1, (const uint8_t*)(blocks + i), sizeof(uint32_t));
```

Comment:
isn't the add here also ub then?

shouldn't we first cast blocks to uint8_t, and add i*4?

Reply:
Ah, I was under the impression only the dereferencing was problematic, but according to C11 6.3.2.3.7:

> A pointer to an object type may be converted to a pointer to a different object type. If the
> resulting pointer is not correctly aligned for the referenced type, the behavior is
> undefined.